From 501a604714cc554864ba6a7b7e8cafb6abf2f1b5 Mon Sep 17 00:00:00 2001
From: Yong Chen <yong.a.chen@intel.com>
Date: Mon, 9 Dec 2013 17:02:10 +0800
Subject: [PATCH 2/2] Enable renderscript vectorizer by default

BZ: 156603

Turn on renderscript vectorizer by default for all platforms.
Introduce sysprop rs.x86vectorizer.disable, set 1 or true to
turn off rs vectorizer.

Change-Id: I56593ab46cd7813e30b3742004dd9db5ae42719e
Category: device enablement
Domain: AOSP.Optimization-RenderScript
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Yong Chen <yong.a.chen@intel.com>
---
 lib/Renderscript/RSVectorizationSupport.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/lib/Renderscript/RSVectorizationSupport.cpp b/lib/Renderscript/RSVectorizationSupport.cpp
index 3715fd1..f0b2408 100644
--- a/lib/Renderscript/RSVectorizationSupport.cpp
+++ b/lib/Renderscript/RSVectorizationSupport.cpp
@@ -542,16 +542,17 @@ const char* g_NoInlineBuiltins[] = {
 unsigned int g_NoInlineBuiltinsNum = sizeof(g_NoInlineBuiltins) / sizeof(char*);
 
 /**
- * run vectorizer if debug.rs.x86vectorizer is set.
+ * rs vectorizer is default turned on.
+ * turn off if rs.x86vectorizer.disable is set to 1 or true.
  */
 bool RSVectorizationSupport::isVectorizerEnabled() {
 #ifdef HAVE_ANDROID_OS
   char buf[PROPERTY_VALUE_MAX];
-  property_get("debug.rs.x86vectorizer", buf, "0");
+  property_get("rs.x86vectorizer.disable", buf, "0");
   if ((::strcmp(buf, "1") == 0) || (::strcmp(buf, "true") == 0))
-    return true;
-  else
     return false;
+  else
+    return true;
 #else
   return true;
 #endif
-- 
2.7.4

